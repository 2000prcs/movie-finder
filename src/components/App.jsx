import React, { Component } from 'react';
import {
  Container,
  Header,
  Icon,
  Pagination,
} from 'semantic-ui-react';
import Search from './Search.jsx';
import Movies from './Movies.jsx';

export default class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      totalPages: 10,
      activePage: 1,
      keyword: 'cat',
    };

    this.handlePaginationChange = this.handlePaginationChange.bind(this);
  }

  componentDidMount() {
    this.fetchMovieData();
  }

  fetchMovieData() {
    const key = '403ffcb3b4481da342203f94fb6e937e';
    const keyword = this.state.keyword;
    const page = this.state.activePage;

    fetch(`https://api.themoviedb.org/3/search/movie?api_key=${key}&query=${keyword}&page=${page}`)
      .then(res => res.json())
      .then((data) => {
        console.log('Movie data received :', data);
        this.setState({ movies: data.results });
        this.setState({ totalPages: data.total_pages });
      })
      .catch(error => console.log('Failed', error));
  }


  handleInputChange() {

  }

  handlePaginationChange(e, { activePage }) {
    this.setState({ activePage }, () => {
      this.fetchMovieData();
    });
  }

  render() {
    const { movies, totalPages, activePage } = this.state;

    return (
      <Container>
        <div className="header">
          <Header as="h1" icon textAlign="center">
            <Icon name="video" circular />
            <Header.Content>
                Let's Search Movies!
            </Header.Content>
          </Header>
        </div>
        <Search handleInputChange={this.handleInputChange} />
        <Movies movies={movies} />
        <div>
          <Pagination
            activePage={activePage}
            ellipsisItem={{ content: <Icon name="ellipsis horizontal" />, icon: true }}
            firstItem={{ content: <Icon name="angle double left" />, icon: true }}
            lastItem={{ content: <Icon name="angle double right" />, icon: true }}
            prevItem={{ content: <Icon name="angle left" />, icon: true }}
            nextItem={{ content: <Icon name="angle right" />, icon: true }}
            onPageChange={this.handlePaginationChange}
            totalPages={totalPages}
          />
        </div>
      </Container>
    );
  }
}
