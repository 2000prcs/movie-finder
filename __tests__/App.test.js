import React from 'react';
import sinon from 'sinon';
import "isomorphic-fetch";
import App from '../src/components/App.jsx';


describe('App component', () => {
  test('Should render App', () => {
    const AppWrapper = shallow(<App />);
    expect(AppWrapper).toMatchSnapshot();
  });

  test('Should fetch popular movies data from API upon mounting', () => {
    sinon.spy(App.prototype, 'fetchMovieData');
    const AppWrapper = mount(<App />);
    expect(App.prototype.fetchMovieData.calledOnce).toBe(true);
    App.prototype.fetchMovieData.restore();
    AppWrapper.unmount();
  });

  test('Should fetch movies data from API when user types a keyword', () => {
    const AppWrapper = mount(<App />);
    AppWrapper.setState({ activePage: 1, keyword: 'cat' });
    AppWrapper.instance().fetchMovieData(1, 'cat');
    AppWrapper.update();
    console.log(AppWrapper.state());
    expect(AppWrapper.state().movie).toBe(true);
    AppWrapper.unmount();
  });

});