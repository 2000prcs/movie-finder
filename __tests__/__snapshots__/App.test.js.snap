// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Should render App 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="header"
        >
          <Header
            as="h1"
            icon={true}
            textAlign="center"
          >
            <Icon
              as="i"
              circular={true}
              name="video"
            />
            <HeaderContent>
              Mo's Movie Finder
            </HeaderContent>
          </Header>
        </div>,
        <Search
          handleInputChange={[Function]}
          keyword=""
        />,
        <Movies />,
        <div
          className="pagination"
        >
          <Pagination
            activePage={1}
            aria-label="Pagination Navigation"
            boundaryRange={1}
            ellipsisItem={
              Object {
                "content": <Icon
                  as="i"
                  name="ellipsis horizontal"
                />,
                "icon": true,
              }
            }
            firstItem={
              Object {
                "content": <Icon
                  as="i"
                  name="angle double left"
                />,
                "icon": true,
              }
            }
            lastItem={
              Object {
                "content": <Icon
                  as="i"
                  name="angle double right"
                />,
                "icon": true,
              }
            }
            nextItem={
              Object {
                "content": <Icon
                  as="i"
                  name="angle right"
                />,
                "icon": true,
              }
            }
            onPageChange={[Function]}
            pageItem={Object {}}
            prevItem={
              Object {
                "content": <Icon
                  as="i"
                  name="angle left"
                />,
                "icon": true,
              }
            }
            siblingRange={1}
            totalPages={10}
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Header
            as="h1"
            icon={true}
            textAlign="center"
          >
            <Icon
              as="i"
              circular={true}
              name="video"
            />
            <HeaderContent>
              Mo's Movie Finder
            </HeaderContent>
          </Header>,
          "className": "header",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "h1",
            "children": Array [
              <Icon
                as="i"
                circular={true}
                name="video"
              />,
              <HeaderContent>
                Mo's Movie Finder
              </HeaderContent>,
            ],
            "icon": true,
            "textAlign": "center",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "i",
                "circular": true,
                "name": "video",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Mo's Movie Finder",
              },
              "ref": null,
              "rendered": "Mo's Movie Finder",
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleInputChange": [Function],
          "keyword": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "movies": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Pagination
            activePage={1}
            aria-label="Pagination Navigation"
            boundaryRange={1}
            ellipsisItem={
              Object {
                "content": <Icon
                  as="i"
                  name="ellipsis horizontal"
                />,
                "icon": true,
              }
            }
            firstItem={
              Object {
                "content": <Icon
                  as="i"
                  name="angle double left"
                />,
                "icon": true,
              }
            }
            lastItem={
              Object {
                "content": <Icon
                  as="i"
                  name="angle double right"
                />,
                "icon": true,
              }
            }
            nextItem={
              Object {
                "content": <Icon
                  as="i"
                  name="angle right"
                />,
                "icon": true,
              }
            }
            onPageChange={[Function]}
            pageItem={Object {}}
            prevItem={
              Object {
                "content": <Icon
                  as="i"
                  name="angle left"
                />,
                "icon": true,
              }
            }
            siblingRange={1}
            totalPages={10}
          />,
          "className": "pagination",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "activePage": 1,
            "aria-label": "Pagination Navigation",
            "boundaryRange": 1,
            "ellipsisItem": Object {
              "content": <Icon
                as="i"
                name="ellipsis horizontal"
              />,
              "icon": true,
            },
            "firstItem": Object {
              "content": <Icon
                as="i"
                name="angle double left"
              />,
              "icon": true,
            },
            "lastItem": Object {
              "content": <Icon
                as="i"
                name="angle double right"
              />,
              "icon": true,
            },
            "nextItem": Object {
              "content": <Icon
                as="i"
                name="angle right"
              />,
              "icon": true,
            },
            "onPageChange": [Function],
            "pageItem": Object {},
            "prevItem": Object {
              "content": <Icon
                as="i"
                name="angle left"
              />,
              "icon": true,
            },
            "siblingRange": 1,
            "totalPages": 10,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="header"
          >
            <Header
              as="h1"
              icon={true}
              textAlign="center"
            >
              <Icon
                as="i"
                circular={true}
                name="video"
              />
              <HeaderContent>
                Mo's Movie Finder
              </HeaderContent>
            </Header>
          </div>,
          <Search
            handleInputChange={[Function]}
            keyword=""
          />,
          <Movies />,
          <div
            className="pagination"
          >
            <Pagination
              activePage={1}
              aria-label="Pagination Navigation"
              boundaryRange={1}
              ellipsisItem={
                Object {
                  "content": <Icon
                    as="i"
                    name="ellipsis horizontal"
                  />,
                  "icon": true,
                }
              }
              firstItem={
                Object {
                  "content": <Icon
                    as="i"
                    name="angle double left"
                  />,
                  "icon": true,
                }
              }
              lastItem={
                Object {
                  "content": <Icon
                    as="i"
                    name="angle double right"
                  />,
                  "icon": true,
                }
              }
              nextItem={
                Object {
                  "content": <Icon
                    as="i"
                    name="angle right"
                  />,
                  "icon": true,
                }
              }
              onPageChange={[Function]}
              pageItem={Object {}}
              prevItem={
                Object {
                  "content": <Icon
                    as="i"
                    name="angle left"
                  />,
                  "icon": true,
                }
              }
              siblingRange={1}
              totalPages={10}
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Header
              as="h1"
              icon={true}
              textAlign="center"
            >
              <Icon
                as="i"
                circular={true}
                name="video"
              />
              <HeaderContent>
                Mo's Movie Finder
              </HeaderContent>
            </Header>,
            "className": "header",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "h1",
              "children": Array [
                <Icon
                  as="i"
                  circular={true}
                  name="video"
                />,
                <HeaderContent>
                  Mo's Movie Finder
                </HeaderContent>,
              ],
              "icon": true,
              "textAlign": "center",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "as": "i",
                  "circular": true,
                  "name": "video",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Mo's Movie Finder",
                },
                "ref": null,
                "rendered": "Mo's Movie Finder",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleInputChange": [Function],
            "keyword": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "movies": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Pagination
              activePage={1}
              aria-label="Pagination Navigation"
              boundaryRange={1}
              ellipsisItem={
                Object {
                  "content": <Icon
                    as="i"
                    name="ellipsis horizontal"
                  />,
                  "icon": true,
                }
              }
              firstItem={
                Object {
                  "content": <Icon
                    as="i"
                    name="angle double left"
                  />,
                  "icon": true,
                }
              }
              lastItem={
                Object {
                  "content": <Icon
                    as="i"
                    name="angle double right"
                  />,
                  "icon": true,
                }
              }
              nextItem={
                Object {
                  "content": <Icon
                    as="i"
                    name="angle right"
                  />,
                  "icon": true,
                }
              }
              onPageChange={[Function]}
              pageItem={Object {}}
              prevItem={
                Object {
                  "content": <Icon
                    as="i"
                    name="angle left"
                  />,
                  "icon": true,
                }
              }
              siblingRange={1}
              totalPages={10}
            />,
            "className": "pagination",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "activePage": 1,
              "aria-label": "Pagination Navigation",
              "boundaryRange": 1,
              "ellipsisItem": Object {
                "content": <Icon
                  as="i"
                  name="ellipsis horizontal"
                />,
                "icon": true,
              },
              "firstItem": Object {
                "content": <Icon
                  as="i"
                  name="angle double left"
                />,
                "icon": true,
              },
              "lastItem": Object {
                "content": <Icon
                  as="i"
                  name="angle double right"
                />,
                "icon": true,
              },
              "nextItem": Object {
                "content": <Icon
                  as="i"
                  name="angle right"
                />,
                "icon": true,
              },
              "onPageChange": [Function],
              "pageItem": Object {},
              "prevItem": Object {
                "content": <Icon
                  as="i"
                  name="angle left"
                />,
                "icon": true,
              },
              "siblingRange": 1,
              "totalPages": 10,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
